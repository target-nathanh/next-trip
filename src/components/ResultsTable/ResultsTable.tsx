import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { Departure } from '../../types';
import { Container, makeStyles, Tooltip } from '@material-ui/core';
import Icon from '@material-ui/core/Icon';

interface ResultsTableProps {
  departures?: Departure[];
}

const useStyles = makeStyles({
  '@keyframes fadeInAndOut': {
    '0%': {
      opacity: 1,
    },
    '50%': {
      opacity: 0,
    },
    '100%': {
      opacity: 1,
    },
  },
  signalIcon: {
    animation: `$fadeInAndOut 2s infinite`,
    paddingLeft: '5px',
  },
  departureTextCell: {
    display: 'flex',
    alignItems: 'center',
  },
  departureText: {
    paddingLeft: '10px',
  },
  table: {
    margin: '10px',
  },
  headerCell: {
    fontSize: '16px',
  },
});

const ResultsTable: React.FC<ResultsTableProps> = ({ departures }) => {
  const classes = useStyles();
  const hasDepartures = departures && departures.length > 0;
  return (
    <Container>
      {!hasDepartures && <div>No departures at this time</div>}
      {hasDepartures && (
        <TableContainer className={classes.table}>
          <Table size="small">
            <TableHead>
              <TableRow>
                <TableCell className={classes.headerCell}>Route</TableCell>
                <TableCell className={classes.headerCell}>Destination</TableCell>
                <TableCell className={classes.headerCell}>Departs</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {departures?.map((departure) => (
                <TableRow key={departure.trip_id}>
                  <TableCell component="th" scope="row">
                    {departure.route_short_name}
                  </TableCell>
                  <TableCell>{departure.description}</TableCell>
                  <TableCell className={classes.departureTextCell}>
                    {departure.departure_text}
                    {departure.actual && (
                      <Tooltip title="Live result">
                        <Icon className={classes.signalIcon} aria-label={'Live result'}>
                          wifi
                        </Icon>
                      </Tooltip>
                    )}
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      )}
    </Container>
  );
};

export default ResultsTable;
