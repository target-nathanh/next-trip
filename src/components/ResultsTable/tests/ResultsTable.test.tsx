import React from 'react';
import { render } from '@testing-library/react';
import ResultsTable from '../ResultsTable';
import results from '../../../mockData/nexTripResultMock.json';
import { MemoryRouter } from 'react-router-dom';

describe('Results', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <ResultsTable departures={results.departures} />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
