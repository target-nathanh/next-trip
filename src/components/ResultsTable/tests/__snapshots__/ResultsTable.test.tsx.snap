// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results matches snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    class="MuiTableContainer-root makeStyles-table-5"
  >
    <table
      class="MuiTable-root"
    >
      <thead
        class="MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head makeStyles-headerCell-6 MuiTableCell-sizeSmall"
            scope="col"
          >
            Route
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head makeStyles-headerCell-6 MuiTableCell-sizeSmall"
            scope="col"
          >
            Destination
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head makeStyles-headerCell-6 MuiTableCell-sizeSmall"
            scope="col"
          >
            Departs
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            122
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            University AV/Circulator
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            5:54
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Franklin Av / To Hennepin
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            14 Min
            <span
              aria-hidden="true"
              aria-label="Live result"
              class="material-icons MuiIcon-root makeStyles-signalIcon-2"
              title="Live result"
            >
              wifi
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            6
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            27Av-Univ / Via Xerxes
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            21 Min
            <span
              aria-hidden="true"
              aria-label="Live result"
              class="material-icons MuiIcon-root makeStyles-signalIcon-2"
              title="Live result"
            >
              wifi
            </span>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            122
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            University AV/Circulator
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            6:14
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Franklin Av / To Hennepin
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            6:17
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            6
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            27Av-Univ / Via Xerxes
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            6:28
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            122
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            University AV/Circulator
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            6:34
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Franklin Av / To Hennepin
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            6:37
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            122
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            University AV/Circulator
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            6:54
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Franklin Av / To Hennepin
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            6:57
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            6
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            27Av-Univ / Via France
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            7:08
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            122
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            University AV/Circulator
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            7:14
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Franklin Av / To Hennepin
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            7:17
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            2
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            Franklin Av / To Hennepin
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            7:28
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            122
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            University AV/Circulator
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            7:34
          </td>
        </tr>
        <tr
          class="MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
            role="cell"
            scope="row"
          >
            6
          </th>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
          >
            27Av-Univ / Via Wooddale
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body makeStyles-departureTextCell-3 MuiTableCell-sizeSmall"
          >
            7:38
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
